contracts/FundFactory.sol

```diff
@@ -16,28 +16,48 @@ import "./Fund.sol";
  */
 contract FundFactory is Administratable {
   // ========== STATE VARIABLES ==========
+  address public endaomentAdmin;
   Fund[] public createdFunds;
   event FundCreated(address indexed newAddress);
+  event EndaomentAdminChanged(address indexed oldAddress, address indexed newAddress);
 
   // ========== CONSTRUCTOR ==========
   /**
    * @notice Create new Fund Factory
    * @param adminContractAddress Address of EndaomentAdmin contract.
    */
-  constructor(address adminContractAddress) public onlyAdmin(adminContractAddress) {}
+  constructor(address adminContractAddress) public {
+    require(adminContractAddress != address(0), "FundFactory: Admin cannot be the zero address");
+    endaomentAdmin = adminContractAddress;
+    emit EndaomentAdminChanged(address(0), adminContractAddress);
+  }
+
+  /**
+   * @notice Update address of the endaomentAdmin contract
+   * @param newAdmin New address of the endaomentAdmin contract
+   */
+  function updateEndaomentAdmin(address newAdmin) public onlyAdmin(endaomentAdmin) {
+    // Validate that contract has a valid admin address set
+    require(newAdmin != address(0), "FundFactory: New admin cannot be the zero address");
+    EndaomentAdmin endaomentAdminContract = EndaomentAdmin(newAdmin);
+    address admin = endaomentAdminContract.getRoleAddress(IEndaomentAdmin.Role.ADMIN);
```

Comment:
Would reduce SLOC if we impl the above here for example.

Reply:
Sorry not sure I follow. Do you mean the fact this function is repeated in both factories?

Reply:
Nvm I was confused.